code_challenges/adhoc/insert.py

```diff
@@ -0,0 +1,9 @@
+def insert(num, nums):
+    number_amount = len(nums) - 1
+    for i, char in enumerate(nums):
+        if num < nums[i]:
+            nums[i:i] = [num]
+            break
+        if i == number_amount:
+            nums.append(num)
+            break
```

Comment:
`char` is unused and poorly named.  You could rename it something like `num_from_nums` and use it on line 4.

Nice find on line 5, `nums[i:i] = [num]`.  This is called  `slice assignment`.  Since this is only doing it one time it is basically the same as the insert was.  The main difference is you are creating a list on the right hand side of the equal.  Creating the list just to add a single item to the other list does take a bit more time than insert though. So if this was a repeated operation insert would be better.